sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -12,19 +12,157 @@
 #include <string>
 
 namespace Azure { namespace Core {
+
+  /**
+   * @brief An exception that gets thrown when @DateTime error occurs.
+   */
+  class DateTimeException : public std::runtime_error {
+  public:
+    /**
+     * @brief Construct with message string.
+     *
+     * @param msg Message string.
+     */
+    explicit DateTimeException(std::string const& msg) : std::runtime_error(msg) {}
+  };
+
   /**
    * @brief Manages date and time in standardized string formats.
    */
   class DateTime {
   public:
-    /// A type that represents tick spans.
-    typedef uint64_t IntervalType;
-
-  private:
-    //  Number of seconds between 01-01-1970 and 01-01-1601.
-    static constexpr IntervalType WindowsToPosixOffsetSeconds = 11644473600LL;
+    /**
+     * @brief Represents time duration.
+     */
+    class Duration {
+      friend class DateTime;
+
+      int64_t m_100nsIntervals;
+      constexpr explicit Duration(int64_t intervalsOf100ns) : m_100nsIntervals(intervalsOf100ns) {}
```

Comment:
Do we need to expose the resolution in the argument name? What if we ant to increase that in the future?